tools/wpt/browser.py

```diff
@@ -939,44 +944,214 @@ class Chrome(ChromeChromiumBase):
+        return self._get_webdriver_url_by_build(version)
+
+
+    def _get_webdriver_url_by_build(self, version):
+        """Find a ChromeDriver download URL based on a MAJOR.MINOR.BUILD version.
+
+        Returns: Download URL string or None if no matching build is found.
+        """
+        version = self._get_build_version(version)
         try:
-            latest = get(latest_url).text.strip()
-        except requests.RequestException:
-            latest_url = f"https://chromedriver.storage.googleapis.com/LATEST_RELEASE_{parts[0]}"
-            try:
-                latest = get(latest_url).text.strip()
-            except requests.RequestException:
-                # We currently use the latest Chromium revision to get a compatible Chromedriver
-                # version for Chrome Dev, since it is not available through the ChromeDriver API.
-                # If we've gotten to this point, it is assumed that this is Chrome Dev.
-                filename = f"chromedriver_{self._chromedriver_platform_string}.zip"
-                revision = self._get_chromium_revision(filename, version)
-                return self._build_snapshots_url(revision, filename)
-        return f"https://chromedriver.storage.googleapis.com/{latest}/{filename}"
+            # Get a list of builds with download URLs from Chrome for Testing.
+            resp = get(
+                "https://googlechromelabs.github.io"
+                "/chrome-for-testing/latest-patch-versions-per-build-with-downloads.json")
+        except requests.RequestException as e:
+            raise requests.RequestException(
+                "Chrome for Testing versions not found", e)
+        builds_json = resp.json()
+        builds_dict = builds_json["builds"]
+        if version not in builds_dict:
+            self.logger.info(f"No builds found for version {version}.")
+            return None
+        download_info = builds_dict[version]["downloads"]
+        if "chromedriver" not in download_info:
+            self.logger.info(f"No ChromeDriver download found for build {version}")
+            return None
+        downloads_for_platform = [d for d in download_info["chromedriver"]
+                                  if d["platform"] == self._chrome_platform_string]
+        if len(downloads_for_platform) == 0:
+            self.logger.info(f"No ChromeDriver download found for build {version}"
+                            f"of platform {self.platform}")
+            return None
+        return downloads_for_platform[0]["url"]
 
-    def download(self, dest=None, channel=None, rename=None):
-        raise NotImplementedError("Downloading of Chrome browser binary not implemented.")
+    def _get_download_urls_by_version(self, version):
+        """Find Chrome for Testing and ChromeDriver download URLs matching a specific version.
+
+        Returns: Both binary downloads for Chrome and ChromeDriver.
```

Comment:
Add a raises section to the docstring for DownloadNotFoundError and requests.RequestException

Reply:
Added to docstring :)